atoma-proxy/src/server/mod.rs

```diff
@@ -0,0 +1,220 @@
+use std::sync::Arc;
+
+use atoma_state::{types::AtomaAtomaStateManagerEvent, AtomaStateManager};
+use atoma_sui::client::AtomaSuiClient;
+use axum::{
+    extract::State,
+    http::{HeaderMap, StatusCode},
+    routing::post,
+    Json, Router,
+};
+use flume::Sender;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tokio::{
+    net::TcpListener,
+    sync::{oneshot, RwLock},
+};
+use tracing::error;
+
+mod config;
+pub use config::AtomaServiceConfig;
+
+use crate::sui::Sui;
+
+const CHAT_COMPLETIONS_PATH: &str = "/v1/chat/completions";
+const NODE_PUBLIC_ADDRESS_REGISTRATION: &str = "/node/registration";
+
+#[derive(Clone)]
+pub struct ProxyState {
+    pub _client: Arc<RwLock<AtomaSuiClient>>,
+    pub state_manager_sender: Sender<AtomaAtomaStateManagerEvent>,
+    pub sui: Arc<RwLock<Sui>>,
+}
+
+fn check_auth(headers: &HeaderMap) -> bool {
```

Comment:
Is this a work in progress ? I presume so, as checking that `auth` is a hardcoded value is not of value

Reply:
since this proxy will manage only single client, we can just use password that can be stored in the config.